jsonrpc/server.go

```diff
@@ -422,8 +422,17 @@ func isBatch(reader *bufio.Reader) bool {
 	return false
 }
 
-func isNil(i any) bool {
-	return i == nil || reflect.ValueOf(i).IsNil()
+func isNilOrEmpty(i any) (bool, error) {
+	if utils.IsNil(i) {
+		return true, nil
+	}
+
+	switch reflect.TypeOf(i).Kind() {
+	case reflect.Slice, reflect.Array, reflect.Map:
+		return reflect.ValueOf(i).Len() == 0, nil
+	default:
+		return false, fmt.Errorf("impossible param type: check request.isSane")
+	}
```

Comment:
It would be great if you could add a test for this case. I think you already describe an example for it below.